baselines/fedexp/fedexp/client.py

```diff
@@ -0,0 +1,120 @@
+"""Client for FedExp."""
+
+import copy
+from collections import OrderedDict
+from typing import Callable, Dict, List, Tuple
+
+import flwr as fl
+import torch
+from flwr.common import NDArrays, Scalar
+from hydra.utils import instantiate
+from omegaconf import DictConfig
+from torch.nn.utils import parameters_to_vector
+from torch.utils.data import DataLoader
+from torchvision.transforms import transforms
+
+
+class FlowerClient(fl.client.NumPyClient):
+    """Standard Flower client for CNN training."""
+
+    def __init__(
+        self,
+        cid: int,
+        net: torch.nn.Module,
+        train_loader: DataLoader,
+        num_epochs: int,
+        data_ratio,
+    ):  # pylint: disable=too-many-arguments
+        print(f"Initializing Client {cid}")
+        self.cid = cid
+        self.train_loader = train_loader
+        self.device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+        self.net = net.to(self.device)
+        self.num_epochs = num_epochs
+        self.data_ratio = data_ratio
+
+    def _set_parameters(self, parameters: NDArrays) -> None:
+        params_dict = zip(self.net.state_dict().keys(), parameters)
+        state_dict = OrderedDict({k: torch.Tensor(v) for k, v in params_dict})
+        self.net.load_state_dict(state_dict, strict=True)
+
+    def fit(
+        self, parameters: NDArrays, config: Dict[str, Scalar]
+    ) -> Tuple[NDArrays, int, Dict]:
+        """Train the network on the training set."""
+        self._set_parameters(parameters)
+        print(f"Client {self.cid} Training...")
+        prev_net = copy.deepcopy(self.net)
+        criterion = torch.nn.CrossEntropyLoss()
+        optimizer = torch.optim.SGD(
+            self.net.parameters(),
+            lr=config["eta_l"],
+            momentum=0,
+            weight_decay=config["weight_decay"],
+        )
+
+        self.net.train()
+        counter = 0
+        while True:
+            for batch_idx, (images, labels) in enumerate(self.train_loader):
+                images, labels = images.to(self.device), labels.to(self.device)
+                if config["use_data_augmentation"]:
+                    transform_train = transforms.Compose(
+                        [
+                            transforms.RandomCrop(32, padding=4),
+                            transforms.RandomHorizontalFlip(),
+                        ]
+                    )
+                    images = transform_train(images)
```

Comment:
having this in the training loop is a bit odd. See my suggestion to do this directly in the `load_datasets()` function, which also requires making a small change to the configs.

Reply:
Please remote the `if` statement.